src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -249,8 +240,29 @@ public override RazorLanguageKind GetLanguageKind(RazorCodeDocument codeDocument
                 }
             }
 
+            // Use the language of the last classified span if we're at the end
+            // of the document.
+            if (classifiedSpans.Count != 0 && absoluteIndex == documentLength)
+            {
+                return GetLanguageFromClassifiedSpan(classifiedSpans.Last());
```

Comment:
Separate the classified span into its own variable to ease debuggability.

Reply:
```suggestion
                var lastClassifiedSpan = classifiedSpans.Last();
                return GetLanguageFromClassifiedSpan(lastClassifiedSpan);
```